Architect-Coders/MU8
# Issue #31: Create Room Database & Issue #33: Save Characters from the service to the Database

**What's in:**

- Initial setup for room library.
- Fixed tech debt on Categories Activity.
- Added entity, DAO and implementation to get Characters from the database if they are already cached from the service.

**Testing Done:**

* On real device: Pixel 2 XL.

**How it looks:** http://bit.ly/2srCrIL

**How to test:**

* Run the app and reach the main screen.
* Tap on the Characters category to enter the screen.
* Notice the delay while the app is downloading 100 characters.
* Get back to the Main Screen after it has loaded and enter again to the Characters Screen, notice how fast it is the load now that we're getting the info from the database.

**Signed-off-by:**

* Diego Recalde <difereto89@gmail.com>
